L1Trigger/L1TTwinMux/plugins/L1TTwinMuxProducer.cc

```diff
@@ -0,0 +1,100 @@
+//-------------------------------------------------
+//
+//   Class: L1TwinMuxProducer
+//
+//   L1TwinMuxProducer EDProducer
+//
+//
+//   Author :
+//   G. Flouris               U Ioannina    Feb. 2015
+//   Mod.: g Karathanasis 
+//--------------------------------------------------
+
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include <FWCore/Framework/interface/ConsumesCollector.h>
+#include <FWCore/Framework/interface/one/EDProducer.h>
+#include <FWCore/ParameterSet/interface/ParameterSet.h>
+
+#include "L1Trigger/L1TTwinMux/interface/L1TTwinMuxAlgorithm.h"
+
+#include <iostream>
+#include <iomanip>
+#include <memory>
+
+using namespace std;
+
+class L1TTwinMuxProducer: public edm::one::EDProducer<edm::one::SharedResources> {
```

Comment:
Why does this need to be based on `edm::one::EDProducer<edm::one::SharedResources>` ?
Can it be made an `edm::stream::EDProducer<>`? This could improve the performance.

There is also no `usesResource` function call in the constructor so in principle this should throw an exception.